### ISSUE TITLE
Unable to get opa istio plugin with policy bundles

### ISSUE THREAD

**Issue by kyle-burnette** [open-policy-agent/opa#2916]
There is a good chance of me missing something as I am relatively still a novice with OPA. 
I have managed to get OPA plugin for Istio fine - so long as I use Istio v 1.6.13 ( version less than 1.7). With that setup I had the policy specified in a config map (likewise is hard coded). 
I wanted to move up and setup policy bundles on a HTTP server to have the ability for OPA to download new/updated policies periodically. 
I followed instructions for setting up bundles on [opa-envoy-plugin for Istio](https://github.com/open-policy-agent/opa-envoy-plugin/tree/master/examples/istio) - Bundles section towards the end as well as the [Bundles documentation on OPA](https://www.openpolicyagent.org/docs/latest/management/#bundles) website.

I created a simple (passive) HTTP server with following folder structure. 
```
/bundles
      /istio
          /authz
               - policy.rego
               - opa-policy-bundle.tar.gz
/opa-policy-bundle.tar.gz 
```

I tested that it is possible to download any of the above files. However I am not entirely clear where the various files should be placed and how exactly would OPA make a request.  The example of `curl -s -L http://mypolicyserver/bundles/istio/authz` is confusing since that just returns the directory contents and not the actual tar.gz file. Note there is no specific file name at the end in the URL. 
My http server is available on a IP address that does not have a DNS entry. 

I then attempted deploying opa-istio using the attached yaml where in the opa-istio-config I have specified the address of the bundle server + port. 
[qs_policy_bundle.yaml.txt](https://github.com/open-policy-agent/opa/files/5556449/qs_policy_bundle.yaml.txt)

It seems to have deployed fine - at least no errors.

Enabled istio proxy and opa-istio injection in default ns. Deployed the sample bookinfo app. However none of its pods would start even after waiting more than an hour at least. Since there were no pods I cannot get any logs from any containers. opa admission controller pods show no errors. 

Thinking there is no ServiceEntry I attempted creating one (below) but am having issues. First, it complained "location" was unrecognized so commented it out. Then am missing some keyword to tell the host is a raw IP addr and not a DNS becoz it tries to interpret it as a DNS and complains. Though I don't know if a ServiceEntry is required since I have specified an externally accessible IP address .  

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: opa-bundle
spec:
  hosts:
    - 52.252.243.18
  ports:
    - number: 9900
      name: http
      protocol: HTTP
      # resolution: DNS
      # location: MESH_EXTERNAL
```

What could I be missing? Thanks for any help.

**Comment by kyle-burnette** [comment]
I tried updating the opa-istio-config to configure the path from where OPA should get the policy bundle more specific (by adding a resource) using the below but there is no change in behavior (sample app pods won't start). This time I did not create a ServiceEntry resource. 
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: opa-istio-config
data:
  config.yaml: |
    services:
      - name: controller
        url: http://52.252.243.18:9900
    bundles:
      istio/authz:
        service: controller
        resource: bundles/istio/authz/opa-policy-bundle.tar.gz
    plugins:
      envoy_ext_authz_grpc:
        addr: :9191
        path: istio/authz/allow
        dry-run: false
        enable-reflection: false
```

**Comment by ashutosh-narkar** [comment]
With the above config OPA will fetch bundles from `http://52.252.243.18:9900/bundles/istio/authz/opa-policy-bundle.tar.gz`. Are there any logs from the opa container or does a `kubectl describe` provide more information ?

**Comment by kyle-burnette** [comment]
I realized I had left references to opa-policy configmap and it's volume in my deployment spec :-( I rectified/removed those. 
Deployed again. The opa admission controller pod logs did not list any errors. This time the pods for sample book info were actually crashing - as against just being stuck last time around (when I could not get any logs). After examining the log for opa-istio container the following error indicated the problem:  "stat /policy/policy.rego: no such file or directory"  

Turns out I had left out a reference to policy.rego in arguments to opa_container within the "inject-policy" configmap for inject.rego. Second blunder :-( 

I removed/deleted that and redeployed again. Bingo! It worked perfectly.

Sorry for the false alarm and thanks your help.

rgds,
KB
